File: CPP/Offer/13.cpp
// 双指针
class Solution
{
  public:
    void reOrderArray(vector<int> &array)
    {
        for (int i = 0; i < array.size(); i++)
        {
            for (int j = array.size() - 1; j > i; j--)
            {
                if (array[j] % 2 == 1 && array[j - 1] % 2 == 0)
                    swap(array[j], array[j - 1]);
            }
        }
    }
};

// 新创建一个数组，将奇数和偶数分别放进去
class Solution
{
  public:
    void reOrderArray(vector<int> &array)
    {
        vector<int> result;
        int num = array.size();
        for (int i = 0; i < num; i++)
        {
            if (array[i] % 2 == 1)
                result.push_back(array[i]);
        }
        for (int i = 0; i < num; i++)
        {
            if (array[i] % 2 == 0)
                result.push_back(array[i]);
        }
        array = result;
    }
};

File: Offer/Python/13.py
# 双指针
class Solution:
    def reOrderArray(self, array):
        if len(array) > 0:
            for i in range(len(array)):
                for j in range(len(array) - 1, i, -1):
                    if (array[j] & 1 and not array[j - 1] & 1):
                        array[j], array[j - 1] = array[j - 1], array[j]
        return array


if __name__ == '__main__':
    s = Solution()
    k = [1, 2, 3, 4, 5]
    t = s.reOrderArray(k)
    print(t)
